File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.API.Native;
using DevExpress.XtraEditors;
using DevExpress.XtraBars.Ribbon;

namespace RichEditFieldsToValues {
    public partial class Form1 : RibbonForm {
        public Form1() {
            InitializeComponent();

            richEditControl1.LoadDocument("Template.rtf");

            richEditControl1.Options.Fields.HighlightMode = FieldsHighlightMode.Always;
            richEditControl1.Options.Fields.HighlightColor = Color.Yellow;

            List<Employee> list = new List<Employee>();
            list.Add(new Employee("Peter Parker", "Spiderman road 123"));
            
            richEditControl1.Options.MailMerge.DataSource = list;
            richEditControl1.Options.MailMerge.ViewMergedData = true;
        }

        private void richEditControl1_CalculateDocumentVariable(object sender, DevExpress.XtraRichEdit.CalculateDocumentVariableEventArgs e) {
            if (e.VariableName == "dvField") {
                e.Value = "Test";
                e.Handled = true;
            }
        }
        private void FieldsToValues(RichEditControl control) {
            control.Document.UnlinkAllFields();
        }

        private void btnReplaceFields_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            FieldsToValues(richEditControl1);
        }
    }   
}

File: VB/Form1.vb
Imports System
Imports System.Drawing
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraBars.Ribbon

Namespace RichEditFieldsToValues

    Partial Public Class Form1
        Inherits RibbonForm

        Public Sub New()
            InitializeComponent()
            richEditControl1.LoadDocument("Template.rtf")
            richEditControl1.Options.Fields.HighlightMode = FieldsHighlightMode.Always
            richEditControl1.Options.Fields.HighlightColor = Color.Yellow
            Dim list As List(Of Employee) = New List(Of Employee)()
            list.Add(New Employee("Peter Parker", "Spiderman road 123"))
            richEditControl1.Options.MailMerge.DataSource = list
            richEditControl1.Options.MailMerge.ViewMergedData = True
        End Sub

        Private Sub richEditControl1_CalculateDocumentVariable(ByVal sender As Object, ByVal e As CalculateDocumentVariableEventArgs)
            If Equals(e.VariableName, "dvField") Then
                e.Value = "Test"
                e.Handled = True
            End If
        End Sub

        Private Sub FieldsToValues(ByVal control As RichEditControl)
            control.Document.UnlinkAllFields()
        End Sub

        Private Sub btnReplaceFields_ItemClick(sender As Object, e As DevExpress.XtraBars.ItemClickEventArgs) Handles btnReplaceFields.ItemClick
            FieldsToValues(richEditControl1)
        End Sub
    End Class
End Namespace
